apis/Google.Cloud.Logging.Console/Google.Cloud.Logging.Console.Tests/GoogleCloudConsoleFormatterTest.cs

```diff
@@ -152,6 +153,101 @@ public void ConsoleLoggerOptions_OptionChange_IsReloaded()
+
+        var logEntry = new LogEntry<string>(LogLevel.Information, "LogCategory", new EventId(1), "test", exception: null, (state, exception) => state);
+        var writer = new StringWriter { NewLine = "\n" };
+
+        formatter.Write(logEntry, scopeProvider: null, writer);
+
+        var expectedJson = "{\"message\":\"test\",\"category\":\"LogCategory\",\"severity\":\"INFO\"}\n";
+        var actualJson = writer.ToString();
+
+        Assert.Equal(expectedJson, actualJson);
+    }
+
+    [Fact]
+    public void Log_DefaultTraceId_TraceInformationIsNotLogged()
+    {
+        // Set the TraceGoogleCloudProjectId.
+        var formatter = CreateFormatter(new GoogleCloudConsoleFormatterOptions { TraceGoogleCloudProjectId = "my-project-id" });
+
+        // Backup the generator, so that it can be reset after the test.
+        // It is null by default but just to keep it easy to understand.
+        var generator = Activity.TraceIdGenerator;
+
+        // Simulate the condition where the generated `TraceId` is equal to the default value of `ActivityTraceId`.
+        Activity.TraceIdGenerator = () => default;
+        using var activity = new Activity("my-activity").Start();
+
+        // Reset the generator, so that other tests aren't impacted.
+        Activity.TraceIdGenerator = generator;
+        
+        var logEntry = new LogEntry<string>(LogLevel.Information, "LogCategory", new EventId(1), "test", exception: null, (state, exception) => state);
+        var writer = new StringWriter { NewLine = "\n" };
+        
+        formatter.Write(logEntry, scopeProvider: null, writer);
+
+        var expectedJson = "{\"message\":\"test\",\"category\":\"LogCategory\",\"severity\":\"INFO\"}\n";
+        var actualJson = writer.ToString();
+
+        Assert.Equal(expectedJson, actualJson);
+    }
+
+    [Fact]
+    public void Log_TraceInformationIsLogged()
+    {
+        // Trace information is logged, only if:
+        // 1. TraceGoogleCloudProjectId is set.
+        // 2. Activity.Current is not null; and
+        // 3. Activity.Current.TraceId does not have the default value.
+
+        // Set the TraceGoogleCloudProjectId.
+        string traceProjectId = "my-project-id";
+        var formatter = CreateFormatter(new GoogleCloudConsoleFormatterOptions { TraceGoogleCloudProjectId = traceProjectId });
+
+        // Start an activity.
+        using var activity = new Activity("my-activity").Start();
+
+        var logEntry = new LogEntry<string>(LogLevel.Information, "LogCategory", new EventId(1), "test", exception: null, (state, exception) => state);
+        var writer = new StringWriter { NewLine = "\n" };
+
+        formatter.Write(logEntry, scopeProvider: null, writer);
+
+        // JsonWriter writes boolean as JSON literal true or false.
+        // We could use JsonConvert but for tests, direct literal is easy and simpler.
+        var expectedTraceSampled = activity?.Recorded ?? false ? "true" : "false";
```

Comment:
If the activity can be null at this point, we really need to fix the test.

```suggestion
        var expectedTraceSampled = activity.Recorded ? "true" : "false";
```

Reply:
Right. Earlier version of these tests used `ActivitySource` and if source didn't have a listener, it could lead to null activity. Now with the `Activity` constructor and start methods, activity won't be null, so the operator isn't needed. Fixed now.